### PR TITLE
Titus Supervisor component IDL

### PR DESCRIPTION
**Pull request by tbak**
### Description of the Change

Initial version of the Titus Supervisor component IDL.

### CODE REVIEW

File: src/main/proto/netflix/titus/titus_master_api.proto [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    /// Get all known TitusMaster instances.
+    rpc GetMasterInstances (google.protobuf.Empty) returns (MasterInstances) {
+    }
+
+    /// Get TitusMaster instance with the given id.
+    rpc GetMasterInstance (MasterInstanceId) returns (MasterInstance) {
+    }
+
+    /// Requests the TitusMaster instance that handles this request to restart.
+    rpc RestartMasterInstance (google.protobuf.Empty) returns (google.protobuf.Empty) {
```
Comment (corindwyer) [inline src/main/proto/netflix/titus/titus_master_api.proto]: I think this method should be more about giving up leadership versus restarting master.

Comment (tbak) [reply src/main/proto/netflix/titus/titus_master_api.proto]: As discussed offline, this endpoint might be easily abused. Changing it to giving up the leadership only is much safer.

File: src/main/proto/netflix/titus/titus_master_api.proto [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+message MasterInstanceId {
+    string instanceId = 1;
+}
+
+message MasterInstances {
+    repeated MasterInstance instances = 1;
+}
+
+service SupervisorService {
```
Comment (fabiokung) [inline src/main/proto/netflix/titus/titus_master_api.proto]: It looks like some methods on this service are intended to be called on a single instance, and others are routed to the correct instance, regardless of which instance gets the request.

For consistency, should we make them all route to the correct instance? i.e.: change `StopBeingLeader()` to also receive a `MasterInstanceId` as a parameter?

Comment (tbak) [reply src/main/proto/netflix/titus/titus_master_api.proto]: There is no routing. Each Master has full information about all the other instances.

Comment (tbak) [reply src/main/proto/netflix/titus/titus_master_api.proto]: I could remove the `GetMasterInstance(<id>)` method. As there are few master instances, fetching full list should be good enough for all cases.

Comment (fabiokung) [reply src/main/proto/netflix/titus/titus_master_api.proto]: I think I understand this better after reviewing Netflix/titus-control-plane#340. Maybe just add a few comments explaining how we rely on gateway to find the current leader and call these methods directly on it?

**Pull request merged**